### PR TITLE
add custom and packet captures

### PR DESCRIPTION
**Pull request by ajbalogh**
Refactor for a more agnostic capture specification.
The following is what it would be like to specify a capture configuration using the current proposal.

```python
def test_capture(api):
    """Test capture functionality

    Capture should allow for multiple filters.
    The capture.filters is a list of CaptureFilter objects which can be 
    CaptureCustom, CaptureEthernet, CaptureVlan and CaptureIpv4.
    A CaptureCustom object is used when packet header is not available.
    """
    config = api.config()

    # create a capture
    config.ports.port(name='port1').port(name='port2')
    capture = config.captures.capture()[0]
    capture.name = 'capture1'
    capture.port_names = [port.name for port in config.ports]

    # add 2 custom filters
    capture.filters.custom().custom()
    src_mac = capture.filters[0]  # type: snappi.CustomFilter
    src_mac.offset = 0
    src_mac.value = '000100000001'
    src_mac.mask = 'FFFFFFFFFFFF'
    dst_mac = capture.filters[1]  # type: snappi.CustomFilter
    dst_mac.offset = 0
    dst_mac.value = '000200000001'
    dst_mac.mask = 'FFFFFFFFFFFF'
    print(config)

    # add packet filters for inner and outer vlan
    capture.filters.clear()
    _, i_vlan, o_vlan = capture.filters.ethernet().vlan().vlan()
    i_vlan.id.value = '01'
    i_vlan.id.mask = 'FF'
    o_vlan.id.value = '03'
    o_vlan.id.mask = 'FF'
    print(config)

    # add packet filters for a custom packet after eth, ip
    capture.filters.clear()
    eth, _, custom = capture.filters.ethernet().ipv4().custom()
    eth.src.value = '000001000001'
    eth.src.mask = 'FFFFFFFFFFFF'
    custom.offset = 16
    custom.value = '010101'
    custom.mask = 'FFFFFF'
    print(config)
```

### CODE REVIEW

File: capture/capture.yaml [hunk 1]
```diff
@@ -57,64 +55,72 @@ components:
...
-          description: >-
-            The mask to be applied to the filter.
           type: string
-        offset:
-          description: >-
-            The offset in the packet to filter at.
-          type: integer
+          format: hex
+
+    Capture.Ethernet:
```
Comment (ankur-sheth) [inline capture/capture.yaml:98]: Do we want to add all these protocol headers in this file itself?  Will become a fairly long file as we add the other headers.
Also, any particular reason why you've only added Ethernet, VLAN and IPv4?

**Review by ankur-sheth — APPROVED** [review]
See one comment further down.

File: capture/capture.yaml (hunk 1 above)
Comment (ajbalogh) [reply capture/capture.yaml:91]: The protocol headers can be moved to separate files once the proposal is accepted.  I've only added ethernet, vlan, ipv4 and custom to illustrate how the capture.filters creation is similar to flow.packet.  Adding additional protocols is a trivial exercise, and an extension like x-replace can be introduced so we can reuse the flow.packet headers and just replace the pattern ref with the capture field ref.

**Comment by ashutshkumr** [comment]
```python
a, b = capture.filters.custom().custom()
```

Does this filter `a && b` or `a || b` ?

**Review by ankur-sheth — APPROVED** [review]

**Comment by ashutshkumr** [comment]
This looks really neat and simple. Although, since we're refactoring capture, I would suggest considering inclusion of following:
- packet count
- capture size of packet

Equivalent to `tcpdump -c 2 -s 100` (2 packets each of slice size 100B). Whether or not we need first 2 or last 2 packets is guided by `overwrite`.

P.S. I was also looking for a way to use boolean operator other than `&&`, but I think this model is in shape so that it can be expanded further later when required.

**Review by ashutshkumr — APPROVED** [review]

**Pull request merged**